docs/en/query_language/functions/higher_order_functions.md

```diff
@@ -119,41 +119,103 @@ SELECT arrayReverseFill(x -> not isNull(x), [1, null, 3, 11, 12, null, null, 5,
 
 Note that the first argument (lambda function) can't be omitted in the `arrayReverseFill` function.
 
-### arraySplit(func, arr1, ...) {#higher_order_functions-array-split}
+### arraySplit {#array-split}
 
-Split `arr1` into multiple arrays. When `func` returns something other than 0, the array will be split on the left hand side of the element. The array will not be split before the first element.
+Creates an array of elements split into groups the length of size (return value from `func`function). When `func` returns something other than 0, the array will be split on the left hand side of the element. If array can't be split evenly, the final chunk will be the remaining elements.
```

Comment:
* mentioning `func` here looks weird because it's defined only later in Syntax section
* from examples below it looks like it returns boolean 1/0 which means whether we should split or not, not the size/length

Reply:
also [1,2,3,4,5] is not so good example input array, because it's a special case when values and indexes are the same which can add some confusion

Reply:
indexes like with 0?